low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,499 @@
+
+    # Dependency list
+    DEPENDENCIES = {"plugins": ["SeviceMsgHandler"]}
+
+    @staticmethod
+    def name():
+        """@return: name of the module."""
+        return ServiceMonitor.SENSOR_NAME
+
+    def __init__(self):
+        """Initialize the relavent datastructures."""
+        super(ServiceMonitor, self).__init__(self.SENSOR_NAME,
+                                                self.PRIORITY)
+
+        Conf.load(self.SSPL_CONFIG, "yaml:///etc/sspl.conf")
+        self.services_to_monitor = \
+            Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.MONITORED_SERVICES}", [])
+
+        self.not_active_services = {}
+        self.failed_services = []
+
+        self.service_status = {}
+
+        self.thread_sleep = \
+            int(Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.THREAD_SLEEP}", 1))
+
+        self.polling_frequency = \
+            int(Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.POLLING_FREQUENCY}", 30))
+
+        self.max_wait_time = \
+            int(Conf.get(self.SSPL_CONFIG, f"{self.SERVICEMONITOR}>{self.MAX_WAIT_TIME}", 60))
+
+    def read_data(self):
+        """Return the dict of service status."""
+        return self.service_status
+
+    def initialize(self, conf_reader, msgQlist, product):
+        """initialize configuration reader and internal msg queues."""
+        # Initialize ScheduledMonitorThread and InternalMsgQ
+        super(ServiceMonitor, self).initialize(conf_reader)
+
+        # Initialize internal message queues for this module
+        super(ServiceMonitor, self).initialize_msgQ(msgQlist)
+
+        # Integrate into the main dbus loop to catch events
+        DBusGMainLoop(set_as_default=True)
+
+        # Initialize SystemBus and get Manager Interface
+        self._bus = SystemBus()
+        systemd = self._bus.get_object("org.freedesktop.systemd1",
+                                       "/org/freedesktop/systemd1")
+        self._manager = Interface(systemd,
+                            dbus_interface='org.freedesktop.systemd1.Manager')
+
+        self.remove_disabled_services()
+
+        return True
+
+    def remove_disabled_services(self):
+        """Remove `disabled` services from the list of services to monitor."""
+        temp = self.services_to_monitor.copy()
+        for service in temp:
+            try:
+                if 'disabled' in str(self._manager.GetUnitFileState(service)):
+                    self.services_to_monitor.remove(service)
+            except DBusException as err:
+                # If a service is enabled then it definitely has 'UnitFileState`,
+                # but for disabled both presence or absence of UnitFileState is
+                # possible. so if `UnitFileState' not present for the service,
+                # it is definitely disabled.
+                logger.error(f"{service} removed from services_to_monitor due "\
+                             f"to error : {err}")
```

Comment:
`Mar 24 00:04:03 localhost sspl-ll[32716]: ERROR hare-consul-agent-c1.service removed from services_to_monitor due to error : org.freedeskt        op.DBus.Error.FileNotFound: No such file or directory (service_monitor.py:128)
`
We get ERROR log for each disabled service if its `UnitFileState` is not found I feel it should be a warning or info.